src/btree/bt_delete.c

```diff
@@ -362,12 +358,11 @@ __wt_delete_page_instantiate(WT_SESSION_IMPL *session, WT_REF *ref)
                 WT_SKIP_FOREACH (ins, insert)
                     ++count;
         }
-        WT_RET(__wt_calloc_def(session, count + 1, &page_del->update_list));
-        __wt_cache_page_inmem_incr(session, page, (count + 1) * sizeof(page_del->update_list));
+        WT_RET(__wt_calloc_def(session, count + 1, &update_list));
```

Comment:
I think we need to increase total size here?

Reply:
This change was deliberate: previously, we incremented the page's cache footprint for the transaction's WT_UPDATE list, but we didn't decrement the footprint when we freed it (at least, not all the time, I don't remember looking if we were consistent). Anyway, now that the WT_UPDATE list is short-lived and discarded when the transaction resolves, I think it's appropriate to leave it off the page's footprint entirely.

Please let me know if you disagree or I've misunderstood some code.

Reply:
The updates will form the bulk of the memory allocated which is tracked, so I'm okay with this.